src/Raven.Server/Web/System/AdminDatabasesHandler.cs

```diff
@@ -1299,5 +1306,115 @@ public Task CompactDatabase()
                 NoContentStatus();
             }
         }
+
+        [RavenAction("/admin/offline-migrate", "POST", AuthorizationStatus.ClusterAdmin)]
+        public async Task MigrateDatabaseOffline()
+        {
+            OfflineMigrationConfiguration configuration;
+            using (ServerStore.ContextPool.AllocateOperationContext(out TransactionOperationContext context))
+            {
+                var migrationConfiguration = await context.ReadForMemoryAsync(RequestBodyStream(), "migration-configuration");
+                configuration = JsonDeserializationServer.OfflineMigrationConfiguration(migrationConfiguration);
+            }
+                
+            var dataDir = configuration.DataDirectory;
+            if(Directory.Exists(dataDir) == false)
+                throw new DirectoryNotFoundException($"Could not find directory {dataDir}");
+
+            //var timeout = GetTimeSpanQueryString("timeout");
+            var dataExporter = configuration.DataExporterFullPath;
+            var databaseName = configuration.DatabaseName;
+            var database = await ServerStore.DatabasesLandlord.TryGetOrCreateResourceStore(databaseName, true);
+            if (database == null)
+            {
+                throw new DatabaseDoesNotExistException($"Can't import into database {databaseName} because it doesn't exist.");
```

Comment:
How is it possible? 2 lines above you get or create database. I assume at this point it should be created.

Reply:
This create the database _instance_, can return null if there is no db record for that name

Reply:
I thought (based on method name: ...OrCreateResourceStore) it always return not null :)

Reply:
I thought so too, but it doesn't.